app/src/main/java/de/westnordost/streetcomplete/quests/building_levels/AddBuildingLevelsForm.kt

```diff
@@ -81,9 +92,7 @@ class AddBuildingLevelsForm : AbstractOsmQuestForm<BuildingLevelsAnswer>() {
     }
 
     override fun isFormComplete() =
-        // levels must be an int >= 0. IF roof levels is specified, it must also be an int >= 0
-        levels?.let { it >= 0 } ?: false
-        && (roofLevels == null || roofLevels?.let { it >= 0 } ?: false)
+        levels != null && (roofsAreUsuallyFlat && !hasNonFlatRoofShape || roofLevels != null)
```

Comment:
I'd always avoid mixing `&&` and `||` on the same level, because you always have to think twice which one takes precedence. So either, I'd add parentheses:

```suggestion
        levels != null && ((roofsAreUsuallyFlat && !hasNonFlatRoofShape) || roofLevels != null)
```

Or, I'd extract a variable:

```kt
private val roofLevelsAreOptional get() = roofsAreUsuallyFlat && !hasNonFlatRoofShape
// …
override fun isFormComplete() =
    levels != null && (roofLevelsAreOptional || roofLevels != null)

Reply:
Really, you have to think twice? Come on, what about `5 + 3 * 2`?

Reply:
But anyway, I like the suggestion with the variable, it is more descriptive. Will do